app/profile/page.tsx

```diff
@@ -1,149 +1,69 @@
-"use client";
-import { AvatarImage, AvatarFallback, Avatar } from "@/components/ui/avatar";
-import { Button } from "@/components/ui/button";
-import { Rating } from "@/components/ui/rating";
-import { FC } from "react";
-import { NavBar } from "@/components/ui/navbar";
-import { SongLayout } from "@/components/ui/song-layout";
-import { mockSongData } from "@/components/mock_data/song_data";
-import { PlaylistLayout } from "@/components/ui/playlist-layout";
-import { mockPlaylistData } from "@/components/mock_data/playlist_data";
-import { mockArtistData } from "@/components/mock_data/artist_data";
-import { mockAirbudsData } from "@/components/mock_data/airbuds_data";
-import Link from "next/link";
-import { useRouter } from 'next/router';
-
-const userData = {
-  name: "Joe Bryant",
-  username: "@joe_bryant",
-  profilePicture: "/profile.jpg",
-  bio: [
-    "Ethan is a freaky frontend enthusiast who has a passion for music and mathematics. He loves to grind, to run around Bel Air and admire the beautiful mansions, and to debone chicken wings. When he has free time he also loves to show his friends his special sauce.",
-    "Other hobbies include: pregaming din tai fung with shake shack, being nonCHALANT, and celebrating the month of May.",
-  ],
-  ratingValue: 4.5,
-  friendsCount: 107,
-};
+'use server';
+import React, { FC, Suspense } from 'react';
+const ProfileSideBar = React.lazy(
+  () => import('@/components/profile_ui/profile-side-bar'),
+);
+import { ProfileProps } from '@/components/profile_ui/profile-side-bar';
+const PinnedSideBar = React.lazy(
+  () => import('@/components/profile_ui/pinned-side-bar'),
+);
+import { SectionProps } from '@/components/profile_ui/pinned-side-bar';
+import { NavBar } from '@/components/navbar';
+import { SkeletonLoader } from '@/components/skeleton_loader';
 
-const Page: FC = () => {
-  const { name, username, profilePicture, bio, ratingValue, friendsCount } = userData;
+//This is the basic function to get the mock data for now
+async function fetchData(endpoint: string) {
+  const res = await fetch(`http://localhost:3000/api/${endpoint}`);
```

Comment:
In nextjs you don't need to prepend `http:localhost:3000` since you're calling it from a server component. The route lives on the same server as the JSX you're rendering. All fetches should look something like `await fetch(`/api/${endpoint}`)

Reply:
https://github.com/zolinthecow/cs35l-big-backs/blob/af9057d80569baccd68b016e0ddb0193968005d8/lib/spotify.ts#L67

Can just do something like this